lms/djangoapps/certificates/signals.py

```diff
@@ -68,16 +68,22 @@ def _listen_for_track_change(sender, user, **kwargs):  # pylint: disable=unused-
 
     user_enrollments = CourseEnrollment.enrollments_for_user(user=user)
     grade_factory = CourseGradeFactory()
+    expected_verification_status, _ = SoftwareSecurePhotoVerification.user_status(user)
     for enrollment in user_enrollments:
         if grade_factory.read(user=user, course=enrollment.course_overview).passed:
-            if fire_ungenerated_certificate_task(user, enrollment.course_id):
-                log.info(u'Certificate generation task initiated for {user} : {course} via track change'.format(
+            if fire_ungenerated_certificate_task(user, enrollment.course_id, expected_verification_status):
+                message = (
+                    u'Certificate generation task initiated for {user} : {course} via track change ' +
+                    u'with verification status of {status}'
+                )
+                log.info(message.format(
                     user=user.id,
-                    course=enrollment.course_id
+                    course=enrollment.course_id,
+                    status=expected_verification_status
                 ))
 
 
-def fire_ungenerated_certificate_task(user, course_key):
+def fire_ungenerated_certificate_task(user, course_key, expected_verification_status=None):
```

Comment:
The code also needs to handle the case when the learner's enrollment track changes.  Did I miss the code for that?